pkg/apis/flow/v1alpha1/jobflow_types.go

```diff
@@ -51,6 +51,16 @@ type DependsOn struct {
 	Targets []string `json:"targets,omitempty"`
 	// +optional
 	Probe *Probe `json:"probe,omitempty"`
+	// +optional
+	TargetsWithStatus []Target `json:"targetsWithStatus,omitempty"`
+}
+
+type Target struct {
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	Name string `json:"name,omitempty"`
+	// +optional
+	Phase v1alpha1.JobPhase `json:"phase,omitempty"`
```

Comment:
Can we add more target states. Such as, liveness probe.
Just an open discussion, this maybe an over-design